### PR TITLE
New1rmchart

### PR DESCRIPTION
**Pull request by Malekai**

### CODE REVIEW

File: app/redux/selectors/SetsSelectors.js [hunk 1]
```diff
@@ -387,10 +387,9 @@ export const getExerciseData = (state, exercise, type) => {
             }       
         });
     } else if (type === 'scatter') {
-        data = [[]]
         sets.forEach((set) => {
             if (set.exercise === exercise && set.reps.length > 0 && set.weight && DateUtils.checkDateWithinRange(range, set.initialStartTime) && (checkIncludesTags(state, set.tags) && checkExcludesTags(state, set.tags))) {
-                data[0].push({ title: set.setID, weight: set.weight, velocity: Number(RepDataMap.averageVelocity(set.reps[0].data)) });
+                data.push({ x: set.weight, y: Number(RepDataMap.averageVelocity(set.reps[0].data)), setID: set.setID });
```
Comment (JonnyLinja) [inline app/redux/selectors/SetsSelectors.js]: - [ ] the first rep isn't necessarily valid and not removed

File: app/features/analysis/cards/OneRM/OneRMView.js [hunk 2]
```diff
@@ -30,21 +30,34 @@ class OneRM extends Component {
...
+        } else {
             return (
                 <View>
-                    <Text style={styles.oneRMText}>e1RM: <Text style={{fontWeight: 'bold'}}>{this.props.e1rm}</Text> {this.props.metric}</Text>
-                    <Text style={{ textAlign: 'center', fontSize: 15 }}>@ <Text style={{ fontWeight: 'bold' }}> {this.props.velocity} m/s</Text></Text> 
-                    <OneRMChart data={this.props.chartData} />
-                    <Text style={styles.confidenceText}>Confidence: {this.props.confidence} %</Text>
+                    <Text style={styles.errorText}>
+                        This exercise has no reps within the date range.
+                    </Text>
```
Comment (JonnyLinja) [inline app/features/analysis/cards/OneRM/OneRMView.js]: - [ ] is this no reps within range, or just not enough? why are you checking for length 3?

File: __tests__/SetsSelectors.test.js [hunk 3]
```diff
@@ -1458,11 +1458,11 @@ describe('SetsSelectors', () => {
...
                 const expected = [
-                    [
-                    {"title": "c", "velocity": 1.73, "weight": 100}, 
-                    {"title": "d", "velocity": 1.83, "weight": 200}, 
-                    {"title": "k", "velocity": 1.43, "weight": 100}]
-                ];
+                    {"setID": "c", "x": 100, "y": 1.73}, 
+                    {"setID": "d", "x": 200, "y": 1.83}, 
+                    {"setID": "k", "x": 100, "y": 1.43}
+                ]
```
Comment (JonnyLinja) [inline __tests__/SetsSelectors.test.js]: - [ ] Go over the regression scenarios so I understand why these values are expected

File: app/features/analysis/cards/OneRM/OneRMView.js (hunk 2 above)
Comment (Malekai) [reply app/features/analysis/cards/OneRM/OneRMView.js]: Not enough reps within the date range, and you need 3 reps to perform a regression (verified by Adam)

File: __tests__/SetsSelectors.test.js (hunk 3 above)
Comment (Malekai) [reply __tests__/SetsSelectors.test.js]: This is identical to the previous chart, just with different keys, the regression scenarios are exactly the same.

**Comment by JonnyLinja** [comment]
- [ ] make a test where first rep is invalid / removed

**Comment by JonnyLinja** [comment]
- [ ] remove the original charting library

File: __tests__/SetsSelectors.test.js [hunk 4]
```diff
@@ -1133,6 +1133,29 @@ describe('SetsSelectors', () => {
                         }],
                         tags: ['F'],
                         initialStartTime: '2018-01-03T04:06:12.640Z'
+                    },
+                                        j: {
```
Comment (JonnyLinja) [inline __tests__/SetsSelectors.test.js]: - [ ] fixs tab spacing

**Pull request merged**